Fix ec2_ami block_device_mapping volume_size to be int in 2.5

mentos1386 opened the pull request:
##### SUMMARY
boto3 added type validation to many methods #28506. Similar to #18999 and #32291, the ec2_ami module didn't cast the volume_size parameter to int before passing it to boto. This was fine if the yaml used an integer value, but if a string was used (because of templating, or variable substitution mostly), a type error was raised.

This #32738 PR will make this PR obsolete. But until then, i think this is a valid fix.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_ami

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.3
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /app/vendor/python3/lib/python3.5/site-packages/ansible
  executable location = /app/vendor/python3/bin/ansible
  python version = 3.5.3 (default, Jan 19 2017, 14:11:04) [GCC 6.3.0 20170118]
```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
If we have a task like this:
```
- ec2_ami:
    name: "{{ ami_name }}"
    state: present
    device_mapping:
      - device_name: /dev/sda1
        volume_size: "{{ ami_size|int }}"
# ...
```
We will get an error that `volume_size` isn't correct type:
```
TASK [ec2_ami] *****************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Invalid type for parameter BlockDeviceMappings[0].Ebs.VolumeSize, value: 10, type: <class 'str'>, valid types: <class 'int'>
```
After applying the PR, there is no issues anymore.

mentos1386 committed:
fix ec2_ami block_device_mapping size to be int

willthames commented on lib/ansible/modules/cloud/amazon/ec2_ami.py:
```diff
@@ -626,13 +626,16 @@ def get_image_by_id(module, connection, image_id):
     new_item = dict_object.get(attribute)
     if new_item is not None:
+        value = dict_object.get(attribute)
+        if type is not None:
```
I don't really like overriding the `type` builtin - can we use `attribute_type` as  parameter or similar?

willthames commented on lib/ansible/modules/cloud/amazon/ec2_ami.py:
```diff
@@ -626,13 +626,16 @@ def get_image_by_id(module, connection, image_id):
+def rename_item_if_exists(dict_object, attribute, new_attribute, child_node=None, type=None):
     new_item = dict_object.get(attribute)
     if new_item is not None:
+        value = dict_object.get(attribute)
```
You're setting `value` exactly the same way that `new_item` was set. Either reuse `new_item` or use `value` instead of `new_item`

mentos1386 committed:
fixed cr issues

renamed `type` to `attribute_type`
reused `new_item` instead of creating new variable `value`

mentos1386 commented:
@willthames Fixed CR issues

willthames commented:
LGTM now, thanks for the fixes - I've asked @s-hertel and @ryansb for a second opinion

willthames commented:
@mentos1386 not that I know of - it'll go out with 2.7 but it's definitely a candidate for backporting to 2.5 and 2.6

To use it prior to release see: http://willthames.github.io/2017/12/12/using-updated-modules-with-stable-ansible.html

willthames commented:
Backport PRs created for 2.5 and 2.6
